### ISSUE TITLE
[C Interface] Garbled NLS subproblem_inform

### ISSUE THREAD

**Issue by jfowkes** [ralna/GALAHAD#305]
As discovered by @AndrewLister-STFC: careful examination of `inform` at the end of `test_nls.py` reveals that `inform['subproblem_inform']` has not been initialised:

```Python
inform['subproblem_inform']
{'status': 0, 'alloc_status': 0, 'bad_alloc': '', 'bad_eval': '                                                                 ', 'iter': 538976288, 'cg_iter': 538976288, 'c_eval':
 538976288, 'j_eval': 538976288, 'h_eval': 538976288, 'factorization_max': 538976288, 'factorization_status': 538976288, 'max_entries_factors': 2314885530818453536, 'factorization_i
nteger': 2314885530818453536, 'factorization_real': 8224, 'factorization_average': 0.0, 'obj': 0.0, 'norm_c': 0.0, 'norm_g': 0.0, 'weight': 0.0, 'time': {'total': 0.0, 'preprocess':
 0.0, 'analyse': 0.0, 'factorize': 0.0, 'solve': 0.0, 'clock_total': 0.0, 'clock_preprocess': 0.0, 'clock_analyse': 0.0, 'clock_factorize': 0.0, 'clock_solve': 0.0}, 'rqs_inform': {
'status': 0, 'alloc_status': 0, 'factorizations': 0, 'max_entries_factors': 0, 'len_history': 0, 'obj': 0.0, 'obj_regularized': 0.0, 'x_norm': 0.0, 'multiplier': 0.0, 'pole': 0.0, '
dense_factorization': False, 'hard_case': False, 'bad_alloc': '', 'time': {'total': 0.0, 'assemble': 0.0, 'analyse': 0.0, 'factorize': 0.0, 'solve': 0.0, 'clock_total': 0.0, 'clock_
assemble': 0.0, 'clock_analyse': 0.0, 'clock_factorize': 0.0, 'clock_solve': 0.0}, 'sls_inform': {'status': 0, 'alloc_status': -1074790400, 'bad_alloc': '', 'more_info': 0, 'entries
': 0, 'out_of_range': 0, 'duplicates': 0, 'upper': 0, 'missing_diagonals': 0, 'max_depth_assembly_tree': 0, 'nodes_assembly_tree': 0, 'real_size_desirable': 0, 'integer_size_desirab
le': 0, 'real_size_necessary': 0, 'integer_size_necessary': 0, 'real_size_factors': 0, 'integer_size_factors': 0, 'entries_in_factors': 0, 'max_task_pool_size': 0, 'max_front_size':
 0, 'compresses_real': 0, 'compresses_integer': 0, 'two_by_two_pivots': 0, 'semi_bandwidth': 0, 'delayed_pivots': 0, 'pivot_sign_changes': 0, 'static_pivots': 0, 'first_modified_piv
ot': 0, 'rank': 0, 'negative_eigenvalues': -1, 'num_zero': -1, 'iterative_refinements': -1, 'flops_assembly': -1, 'flops_elimination': 4294967295, 'flops_blas': 0, 'largest_modified
_pivot': 2.1219957905e-314, 'minimum_scaling_factor': 0.0, 'maximum_scaling_factor': -1.0, 'condition_number_1': -1.0, 'condition_number_2': 0.0, 'backward_error_1': nan, 'backward_
error_2': nan, 'forward_error': nan, 'alternative': True, 'sils_ainfo': {'flag': 0, 'more': -1074790400, 'nsteps': 0, 'nrltot': -1074790400, 'nirtot': 0, 'nrlnec': 0, 'nirnec': -1,
'nrladu': -1, 'niradu': -1, 'ncmpa': -1, 'oor': -1, 'dup': -1, 'maxfrt': -1, 'stat': -1, 'faulty': -1, 'opsa': nan, 'opse': 0.0}, 'sils_finfo': {'flag': 0, 'more': 0, 'maxfrt': 0, '
nebdu': 0, 'nrlbdu': 0, 'nirbdu': 0, 'nrltot': 0, 'nirtot': 0, 'nrlnec': 0, 'nirnec': 0, 'ncmpbr': 0, 'ncmpbi': 0, 'ntwo': 0, 'neig': 0, 'delay': 0, 'signc': 0, 'nstatic': 0, 'modst
ep': 0, 'rank': 0, 'stat': 0, 'faulty': 0, 'step': 0, 'opsa': 0.0, 'opse': -1.0, 'opsb': -1.0, 'maxchange': -1.0, 'smin': -1.0, 'smax': -1.0}, 'sils_sinfo': {'flag': 0, 'stat': -107
4790400, 'cond': 0.0, 'cond2': nan, 'berr': nan, 'berr2': nan, 'error': nan}, 'lapack_error': 0, 'time': {'total': nan, 'analyse': nan, 'factorize': nan, 'solve': nan, 'order_extern
al': -1.0, 'analyse_external': -1.0, 'factorize_external': -1.0, 'solve_external': -1.0, 'clock_total': 0.0, 'clock_analyse': 0.0, 'clock_factorize': nan, 'clock_solve': -1.0, 'cloc
k_order_external': -1.0, 'clock_analyse_external': -1.0, 'clock_factorize_external': -1.0, 'clock_solve_external': -1.0}}, 'ir_inform': {'status': 1, 'alloc_status': 0, 'bad_alloc':
 '', 'norm_initial_residual': 2.1604692836e-314, 'norm_final_residual': 6.013470016999068e-154}}, 'glrt_inform': {'status': 538976288, 'alloc_status': 538976288, 'bad_alloc': '
                                                           ', 'iter': 48, 'iter_pass2': 0, 'obj': 2.4948631008042762e+218, 'obj_regularized': 5e-324, 'multiplier': 0.0, 'xpo_norm':
1.265e-321, 'leftmost': 2.1219957915e-314, 'negative_curvature': True, 'hard_case': True}, 'psls_inform': {'status': 0, 'alloc_status': 0, 'analyse_status': 48, 'factorize_status':
0, 'solve_status': 89, 'factorization_integer': 26, 'factorization_real': 4372838512, 'preconditioner': 77871216, 'semi_bandwidth': 1, 'reordered_semi_bandwidth': 25, 'out_of_range'
: 0, 'duplicates': 2, 'upper': 0, 'missing_diagonals': 58920656, 'semi_bandwidth_used': 1, 'neg1': 20, 'neg2': 0, 'perturbed': True, 'fill_in_ratio': 0.0, 'norm_residual': 0.0, 'bad
_alloc': '', 'sls_inform': {'status': 0, 'alloc_status': -1074790400, 'bad_alloc': '', 'more_info': 0, 'entries': 0, 'out_of_range': 0, 'duplicates': 0, 'upper': 0, 'missing_diagona
ls': 0, 'max_depth_assembly_tree': 0, 'nodes_assembly_tree': 0, 'real_size_desirable': 0, 'integer_size_desirable': 0, 'real_size_necessary': 0, 'integer_size_necessary': 0, 'real_s
ize_factors': 0, 'integer_size_factors': 0, 'entries_in_factors': 0, 'max_task_pool_size': 0, 'max_front_size': 0, 'compresses_real': 0, 'compresses_integer': 0, 'two_by_two_pivots'
: 0, 'semi_bandwidth': 0, 'delayed_pivots': 0, 'pivot_sign_changes': 0, 'static_pivots': 0, 'first_modified_pivot': 0, 'rank': 0, 'negative_eigenvalues': -1, 'num_zero': -1, 'iterat
ive_refinements': -1, 'flops_assembly': -1, 'flops_elimination': 4294967295, 'flops_blas': 0, 'largest_modified_pivot': 2.1219957905e-314, 'minimum_scaling_factor': 0.0, 'maximum_sc
aling_factor': -1.0, 'condition_number_1': -1.0, 'condition_number_2': 0.0, 'backward_error_1': nan, 'backward_error_2': nan, 'forward_error': nan, 'alternative': True, 'sils_ainfo'
: {'flag': 0, 'more': -1074790400, 'nsteps': 0, 'nrltot': -1074790400, 'nirtot': 0, 'nrlnec': 0, 'nirnec': -1, 'nrladu': -1, 'niradu': -1, 'ncmpa': -1, 'oor': -1, 'dup': -1, 'maxfrt
': -1, 'stat': -1, 'faulty': -1, 'opsa': nan, 'opse': 0.0}, 'sils_finfo': {'flag': 0, 'more': 0, 'maxfrt': 0, 'nebdu': 0, 'nrlbdu': 0, 'nirbdu': 0, 'nrltot': 0, 'nirtot': 0, 'nrlnec
': 0, 'nirnec': 0, 'ncmpbr': 0, 'ncmpbi': 0, 'ntwo': 0, 'neig': 0, 'delay': 0, 'signc': 0, 'nstatic': 0, 'modstep': 0, 'rank': 0, 'stat': 0, 'faulty': 0, 'step': 0, 'opsa': 0.0, 'op
se': -1.0, 'opsb': -1.0, 'maxchange': -1.0, 'smin': -1.0, 'smax': -1.0}, 'sils_sinfo': {'flag': 0, 'stat': -1074790400, 'cond': 0.0, 'cond2': nan, 'berr': nan, 'berr2': nan, 'error'
: nan}, 'lapack_error': 538976288, 'time': {'total': nan, 'analyse': nan, 'factorize': nan, 'solve': nan, 'order_external': -1.0, 'analyse_external': -1.0, 'factorize_external': -1.
0, 'solve_external': -1.0, 'clock_total': 0.0, 'clock_analyse': 0.0, 'clock_factorize': nan, 'clock_solve': -1.0, 'clock_order_external': -1.0, 'clock_analyse_external': -1.0, 'cloc
k_factorize_external': -1.0, 'clock_solve_external': -1.0}}, 'time': {'total': nan, 'assemble': nan, 'analyse': nan, 'factorize': nan, 'solve': nan, 'update': nan, 'clock_total': na
n, 'clock_assemble': nan, 'clock_analyse': 0.0, 'clock_factorize': nan, 'clock_solve': nan, 'clock_update': nan}}, 'bsc_inform': {'status': 538976288, 'alloc_status': 538976288, 'ba
d_alloc': '                                ', 'max_col_a': 0, 'exceeds_max_col': 0, 'time': nan, 'clock_time': 0.0}, 'roots_inform': {'status': 2887977, 'alloc_status': 1071644672,
'bad_alloc': '5'}}
```

GDB suggests that it is the C interface that is not initialising/copying over the default Fortran values for subproblem_inform.

**Comment by jfowkes** [comment]
Running our C interface test for NLS (`nlst.c`) reveals that it is indeed the C interface that is garbling `subproblem_inform`:
```
=================================== 96/156 ===================================
test:         GALAHAD:nls+double+C / nlst_c_double
----------------------------------- stdout -----------------------------------
 C sparse matrix indexing

 tests options for all-in-one storage format

 inform.subproblem_inform.factorization_max: 538976288
```
Clearly the value `538976288` is garbage here, similarly for all other `subproblem_inform` values.

**Comment by jfowkes** [comment]
Resolved by 34ffd41
